dashboard/modules/job/tests/test_utils.py

```diff
@@ -0,0 +1,88 @@
+from tempfile import NamedTemporaryFile
+import sys
+
+import pytest
+
+from ray.dashboard.modules.job.utils import file_tail_iterator
+
+
+@pytest.fixture
+def tmp():
+    with NamedTemporaryFile() as f:
+        yield f.name
+
+
+class TestIterLine():
+    def test_invalid_type(self):
+        with pytest.raises(TypeError, match="path must be a string"):
+            next(file_tail_iterator(1))
+
+    def test_file_not_created(self, tmp):
+        it = file_tail_iterator(tmp)
+        assert next(it) is None
+        f = open(tmp, "w")
+        f.write("hi\n")
+        f.flush()
```

Comment:
this reminds me that we're not explicitly calling `flush()` in job_manager to ensure we always capture all logs in buffer when the job terminates ?